### ISSUE TITLE
Incremental indexing broken, related to metas names and compoundExecutor

### ISSUE THREAD

**Issue by JoanFM** [jina-ai/jina#1268]
**Describe the bug**
This is a bug that has been found trying to build an example using `Incremental Indexing`.

We saw that the `integration tests` did not show the expected behavior of the feature, because the `Flow` was never closed before trying to index more documents (which is the intended behavior).

We tried to fix the issue in #1267 but we saw that in many tests we have this exception:

![image](https://user-images.githubusercontent.com/19825685/98841600-a2a68780-2448-11eb-9b07-e377c5902b77.png)

After investigation I saw what is the problem.

When building an index from its `.bin` dump, add some point this is called (Print stacktrace to the call):

```
  File "/home/joan/jina/jina/jina/executors/compound.py", line 308, in _set_routes
    for m, _ in inspect.getmembers(c, predicate=inspect.ismethod):
  File "/usr/lib/python3.7/inspect.py", line 341, in getmembers
    value = getattr(object, key)
  File "/home/joan/jina/jina/jina/helper.py", line 668, in __get__
    value = obj.__dict__[f'CACHED_{self.func.__name__}'] = self.func(obj)
```
The problem is that we are trying to access every `cached_property` of the `executors` because of this `CompoundExecutor` inspection.

What this does is rebuilds the `cached_properties functions` if exists by calling the function. In the `VectorIndexer` case what it does is call the `query_handler`, this query_handler since it has indexed data, sets the `is_handler_loaded` to `True`.

What this does is that when later the indexer tries to get the `write_handler`, it is never obtained because it considers that the indexer is in `QUERY` mode, and since it is forbidden for indexers other than the `DocIDCache` to be in both `QUERY` and `WRITE` mode, it returns None and this error appears.

**Comment by JoanFM** [comment]
**Potential approach**
Maybe we need the concept of a `cached` but not `reconstructed` property?

**Comment by JoanFM** [comment]
Closed via #1267
